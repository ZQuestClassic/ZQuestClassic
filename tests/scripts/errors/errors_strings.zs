#option NO_ERROR_HALT on

void strings() {
	char32 c1 = 'should have been double quotes'; // char too long
	char32 c2 = '\c'; // invalid escape
	char32 c3 = 'should have been double quotes';
	char32 c4 = 'o'; // ok
	char32 c5 = '\n'; // ok
}
