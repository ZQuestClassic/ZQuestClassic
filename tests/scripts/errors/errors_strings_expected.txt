stderr:

Compiling 'errors_strings.zs'
Pass 1: Parsing
ERROR: expected a single character value - did you mean to use double quotes for a string? [errors_strings.zs Line 4 Column 46 ''should have been double quotes'']


2    
3    void strings() {
4        char32 c1 = 'should have been double quotes'; // char too long
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


ERROR: Unknown escape character. [errors_strings.zs Line 5 Column 18 ''\c'']


3    void strings() {
4        char32 c1 = 'should have been double quotes'; // char too long
5        char32 c2 = '\c'; // invalid escape
                     ^~~~


ERROR: expected a single character value - did you mean to use double quotes for a string? [errors_strings.zs Line 6 Column 46 ''should have been double quotes'']


4    char32 c1 = 'should have been double quotes'; // char too long
5    char32 c2 = '\c'; // invalid escape
6    char32 c3 = 'should have been double quotes';
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Failure!
Compile finished with exit code '-1' (compiled with warnings)

stdout:

{
  "success": false,
  "code": -1,
  "diagnostics": [
    {
      "range": {
        "start": {
          "line": 3,
          "character": 13
        },
        "end": {
          "line": 3,
          "character": 45
        }
      },
      "severity": 1,
      "message": "ERROR: expected a single character value - did you mean to use double quotes for a string?"
    },
    {
      "range": {
        "start": {
          "line": 4,
          "character": 13
        },
        "end": {
          "line": 4,
          "character": 17
        }
      },
      "severity": 1,
      "message": "ERROR: Unknown escape character."
    },
    {
      "range": {
        "start": {
          "line": 5,
          "character": 13
        },
        "end": {
          "line": 5,
          "character": 45
        }
      },
      "severity": 1,
      "message": "ERROR: expected a single character value - did you mean to use double quotes for a string?"
    }
  ]
}
